### ISSUE TITLE
Kiam TLS secrets are being created in default namespace not kube-system

### ISSUE THREAD

**Issue by kevtaylor** [kubernetes-incubator/kube-aws#1163]
When the server and agent attempt to start they are missing their TLS secrets as
```
MountVolume.SetUp failed for volume "tls" : secrets "kiam-server-tls" not found
```
On investigation, these secrets are actually in the default namespace and not in the kube-system namespace where the pods get created

**Comment by mumoshu** [comment]
@kevtaylor Thanks for the port!

Could you run `kubectl get secret -n kube-system`?
For me, they're in kube-sytem:

```
 k get secret -n kube-system
NAME                               TYPE                                  DATA      AGE
calico-etcd-secrets                Opaque                                3         13d
default-token-s996h                kubernetes.io/service-account-token   3         13d
heapster-token-h4krq               kubernetes.io/service-account-token   3         13d
kiam-agent-tls                     Opaque                                3         13d
kiam-server-tls                    Opaque                                3         13d
kiam-server-token-qj5xx            kubernetes.io/service-account-token   3         13d
kube-dns-token-75kt9               kubernetes.io/service-account-token   3         13d
kube-proxy-token-ddnkb             kubernetes.io/service-account-token   3         13d
kubernetes-dashboard-certs         Opaque                                0         13d
kubernetes-dashboard-key-holder    Opaque                                2         13d
kubernetes-dashboard-token-r8lgp   kubernetes.io/service-account-token   3         13d
```

**Comment by kevtaylor** [comment]
Interesting - they certainly were seen in the default namespace and not in kube-system and after the fix they were in kube-system. The pods were not finding the secrets and now they do, so not sure how this worked for you.

**Comment by mumoshu** [comment]
Yeah, really interesting..

I've surveyed the upstream issues a bit but found just https://github.com/kubernetes/kubernetes/issues/51068, which seems to have fixed since v1.8. 
Not sure what's going on undert the hood.

But I'm going to merge your fix as it won't break anything!

**Comment by kevtaylor** [comment]
Yes was just going to add that this was the output
```
apiVersion: v1
data:
  kiam-ca.pem: <data>
  kiam-server-key.pem: <data>
  kiam-server.pem: <data>
kind: Secret
metadata:
  creationTimestamp: null
  name: kiam-server-tls
```
And to highlight that this was a k8s 1.8.4 system

**Comment by mumoshu** [comment]
Ah, thx for the info!
It seems like v1.8.4 doesn't include the fix yet: https://github.com/kubernetes/kubernetes/blob/9befc2b8928a9426501d3bf62f72849d5cbcd5a3/pkg/kubectl/cmd/create.go#L291

**Comment by mumoshu** [comment]
Closing this as resolved via #1164
